docs/how_to/use_estimator.rst

```diff
@@ -0,0 +1,145 @@
+
+.. note::
+
+    While this guide only uses Qiskit Terra's implementation of the ``Estimator`` primitive, there are other
+    implementations of this primitive like Qiskit Terra's :class:`~qiskit.primitives.BackendEstimator`, Qiskit Aer's :class:`~qiskit_aer.primitives.Estimator`
+    and Qiskit Runtime's :class:`~qiskit_ibm_runtime.Estimator`.
+
+
+Initialize observable
+=====================
+
+The first step is to define the observable whose expected value you want to compute. This observable can be any ``BaseOperator``, like the operators from :mod:`qiskit.quantum_info`.
+Among them it is preferable to use :class:`~qiskit.quantum_info.SparsePauliOp`.
+
+.. testcode::
+
+    from qiskit.quantum_info import SparsePauliOp
+
+    obs = SparsePauliOp(["II", "XX", "YY", "ZZ"], [1, 1, -1, 1])
+
+Initialize quantum circuits
+===========================
+
+Then you need to create the :class:`~qiskit.circuit.QuantumCircuit` for which you want to obtain the expected value.
+
+.. plot::
+    :include-source:
+
+    from qiskit import QuantumCircuit
+
+    qc = QuantumCircuit(2)
+    qc.h(0)
+    qc.cx(0,1)
+    qc.draw("mpl")
+
+.. testsetup::
+
+    from qiskit import QuantumCircuit
+
+    qc = QuantumCircuit(2)
+    qc.h(0)
+    qc.cx(0,1)
+
+Initialize the ``Estimator``
+============================
+
+Then, you need to create an :class:`~qiskit.primitives.Estimator` object.
+
+.. testcode::
+
+    from qiskit.primitives import Estimator
+
+    estimator = Estimator()
+
+Run and get results
+===================
+
+Now that you have defined ``estimator``, you can create a :class:`~.PrimitiveJob` (subclass of :class:`~qiskit.providers.JobV1`) with the
+:meth:`~qiskit.primitives.Estimator.run` method and, then, you can get the results (as a :class:`~qiskit.primitives.EstimatorResult` object) with
+the results with the :meth:`~qiskit.providers.JobV1.result` method.
+
+.. testcode::
+
+    job = estimator.run(qc, obs)
+    result = job.result()
+    print(result)
+
+.. testoutput::
+
+    EstimatorResult(values=array([4.]), metadata=[{}])
+
+Get the expected value
+----------------------
+
+From these results you can take the expected values with the attribute :attr:`~qiskit.primitives.EstimatorResult.values`.
+
+Generally, :attr:`~qiskit.primitives.EstimatorResult.values` returns a `numpy.ndarray <https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html>`_
+whose ``i``-th element would be the expectation value corresponding to the ``i``-th circuit and ``i``-th observable.
+
+.. testcode::
+
+    exp_value = result.values[0]
+    print(exp_value)
+
+
+.. testoutput::
+
+    3.999999999999999
+
+Parameterized circuits with ``Estimator``
+=========================================
+
+The :class:`~qiskit.primitives.Estimator` primitive also has the option to include unbound parameterized circuits like the one below.
```

Comment:
Here this starts to get close to using an optimizer with a parameterized circuit to find minimum exp val. i.e VQE. Would we want a howto to show using an estimator for that. Of course we have a more comprehensive pre-built VQE algo - but its a nice simple howto to build a simple one. Here the parameters are applied sequentially - rather than all at once in this example.

Reply:
I guess I see people using the Estimator with an optimizer to make a simple VQE. It just seemed more like an aspect someone might be wanting to do and a howto, that includes parameterized circuits, which seemed to be the aspect the latter section was bringing in, might be a task that had more relevancy. I agree its bit more complex than what is there and maybe is more applicable for a future howto. It was less about VQE and more that this a common way to use parameterized ccts with an Estimator.